src/Compilers/CSharp/Portable/Symbols/SubstitutedTypeParameterSymbol.cs

```diff
@@ -96,10 +96,10 @@ public override string Name
             }
         }
 
-        internal override ImmutableArray<TypeWithAnnotations> GetConstraintTypes(ConsList<TypeParameterSymbol> inProgress, bool canIgnoreNullableContext)
+        internal override ImmutableArray<TypeWithAnnotations> GetConstraintTypes(ConsList<TypeParameterSymbol> inProgress, bool canUseLightweightTypeConstraintBinding)
         {
             var constraintTypes = ArrayBuilder<TypeWithAnnotations>.GetInstance();
-            _map.SubstituteConstraintTypesDistinctWithoutModifiers(_underlyingTypeParameter, _underlyingTypeParameter.GetConstraintTypes(inProgress, canIgnoreNullableContext), constraintTypes, null);
+            _map.SubstituteConstraintTypesDistinctWithoutModifiers(_underlyingTypeParameter, _underlyingTypeParameter.GetConstraintTypes(inProgress, canUseLightweightTypeConstraintBinding), constraintTypes, null);
```

Comment:
>canUseLightweightTypeConstraintBinding [](start = 149, length = 38)

I feel like `can` might be the wrong word here. It's not an optional thing: if there are constraints, and we don't use lightweight binding, we'll cause cycles.

Reply:
That scenarios that request full binding may not get lightweight results, but other scenarios can use lightweight binding. So we have two scenarios: (1) require full binding, (2) would accept ("can use") lightweight binding.

---
In reply to: [502585995](https://github.com/dotnet/roslyn/pull/48270#discussion_r502585995) [](ancestors = 502585995)